docs/docs/creating-dynamically-rendered-navigation.md

```diff
@@ -2,7 +2,176 @@
+
+Examining the available types in GraphQL you will come across a site query that looks as follows:
+
+```js
+site(
+    siteMetadata: siteSiteMetadataInputObject_2
+    port: sitePortQueryString_2
+    host: siteHostQueryString_2
+    pathPrefix: sitePathPrefixQueryString_2
+    polyfill: sitePolyfillQueryBoolean_2
+    buildTime: siteBuildTimeQueryString_2
+    id: siteIdQueryString_2
+    internal: siteInternalInputObject_2
+): Site
+```
+At this point, it is useful if you know a little GraphQL in order to create a query which can extract the links from the site meta data. If you are unfamiliar with GraphQL, there is some excellent documentation available at (LINK) so you can brush up on your skills! However you can also use the query below so you can keep following along with this guide until you have time to learn more:
+```js
+query SiteQuery {
+  site {
+    siteMetadata {
+      title
+      menuLinks {
+        name
+        link
+      }
+    }
+  }
+}
+``` 
+When executing this query within the GraphiQL editor you see output that looks similar to the following:
+```js
+{
+  "data": {
+    "site": {
+      "siteMetadata": {
+        "title": "Gatsby Default Starter",
+        "menuLinks": [
+          {
+            "name": "home",
+            "link": "/"
+          },
+          {
+            "name": "page2",
+            "link": "/page-2"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+Perfect! You now have a way of obtaining data from the `gatsby-config.js` file,  lets continue by pulling this data into the layout using the query you just formed.
+
+### Pulling data inside the layout component
+Inside your project locate the `src/components` folder and navigate to the `layout.js` file. Within this layout component you should notice a component named `StaticQuery`. 
+
+StaticQuery is a new component introduced in Gatsby V2 and allows you to run graphql queries within your components, not just pages. It allows developers to colocate data with their components.
+
+Currently the layout component should look like the following:
+```js
+  <StaticQuery
+    query={graphql`
+      query SiteTitleQuery {
+        site {
+          siteMetadata {
+            title
+            menuLinks {
+              name
+              link
+            }
+          }
+        }
+      }
+    `} ...left out for brevity/>
+```
+
+Lets extend the query within this component to include the menuLinks meta data, so it looks as so: (**Note:** *The graphql query must be located within the StaticQuery component, rather than referencing a variable from elsewhere due to how Gatsby parses our projec*t).
+```js
+<StaticQuery query={
+      graphql`
+      query SiteQuery {
+        site {
+          siteMetadata {
+            title
+            menuLinks {
+              name
+              link
+            }
+          }
+        }
+      }
+    `} 
+...left out for brevity />
```

Comment:
This can and previous can be compressed using `diff` codeblock - something we do for example here - https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/migrating-from-v1-to-v2.md#5-change-query-to-use-staticquery

Reply:
Yeah that's a really good idea, was unsure of how to show changes and wasn't aware you could do diff in markdown. I will look into this tomorrow.